src/components/Actions.tsx

```diff
@@ -233,113 +233,130 @@ export const ShapesSwitcher = ({
-            icon={icon}
-            checked={activeTool.type === value}
-            name="editor-current-shape"
-            title={`${capitalizeString(label)} — ${shortcut}`}
-            keyBindingLabel={numericKey || letter}
-            aria-label={capitalizeString(label)}
-            aria-keyshortcuts={shortcut}
-            data-testid={`toolbar-${value}`}
-            onPointerDown={({ pointerType }) => {
-              if (!appState.penDetected && pointerType === "pen") {
-                app.togglePenMode(true);
-              }
-            }}
-            onChange={({ pointerType }) => {
-              if (appState.activeTool.type !== value) {
-                trackEvent("toolbar", value, "ui");
-              }
-              if (value === "image") {
-                app.setActiveTool({
-                  type: value,
-                  insertOnCanvasDirectly: pointerType !== "mouse",
-                });
-              } else {
-                app.setActiveTool({ type: value });
-              }
-            }}
-          />
-        );
-      })}
+            const label = isDefaultShape(value) ? t(`toolBar.${value}`) : value;
+            const letter =
+              key && capitalizeString(typeof key === "string" ? key : key[0]);
+            const shortcut = letter
+              ? `${letter} ${t("helpDialog.or")} ${numericKey}`
+              : `${numericKey}`;
+            return (
+              <ToolButton
+                className={clsx("Shape", { fillable })}
+                key={customType || value}
+                type="radio"
+                icon={icon}
+                checked={
+                  activeTool.type === value &&
+                  activeTool.customType === customType
+                }
+                name="editor-current-shape"
+                title={`${capitalizeString(label)} — ${shortcut}`}
+                keyBindingLabel={numericKey || letter}
+                aria-label={capitalizeString(label)}
+                aria-keyshortcuts={shortcut}
+                data-testid={`toolbar-${value}`}
+                onPointerDown={({ pointerType }) => {
+                  if (!appState.penDetected && pointerType === "pen") {
+                    app.togglePenMode(true);
+                  }
+                }}
+                onChange={({ pointerType }) => {
+                  if (isDefaultShape(value)) {
+                    if (appState.activeTool.type !== value) {
+                      trackEvent("toolbar", value, "ui");
+                    }
+                    if (value === "image") {
+                      app.setActiveTool({
+                        type: value,
+                        insertOnCanvasDirectly: pointerType !== "mouse",
+                      });
+                    } else {
+                      app.setActiveTool({ type: value });
+                    }
+                  } else if (customType) {
+                    app.setActiveTool({ type: "custom", customType });
```

Comment:
What is the reason we add extra nesting
```type: "custom", customType ```

Instead of using our ```customType``` as the type

Reply:
You already created a function is ```isDefaultShape```

Reply:
I did it in order to satisfy the Excalidraw ActiveTool type:

```
export type ActiveTool =
  | {
      type: ToolType;
      customType: null;
    }
  | {
      type: "custom";
      customType: string;
    };
```